wordpress-mobile/WordPress-Android
# [Reader IA] Update selected content stream based on menu item

Part of #19605 

> [!CAUTION]
> ~Should not be merged before https://github.com/wordpress-mobile/WordPress-Android/pull/19805~

`ReaderViewModel` test will be updated in my next PR.

https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/888fc7df-29f0-4e62-a945-c664699fccdf


-----

## To Test:
1 - Open reader
2 - Select menu items in multiple states (e.g. "Saved" with and without saved posts): the posts list should be updated correctly
3 - Make sure you have no saved posts and open "Saved" menu item
4 - Tap on "Go to subscriptions": it should open "Subscriptions" menu item correctly

-----

## Regression Notes

1. Potential unintended areas of impact
- Reader existing fragments

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing

3. What automated tests I added (or what prevented me from doing so)
- Unit tests will be implemented in another PR 

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
